### PR TITLE
ci: aarch64: Fix git bisect false positives and wiki failure

### PR DESCRIPTION
**Pull request by Ryo-not-rio**
- add a fix for the git bisect conflicting with the wiki causing the wiki update to fail
- add a fix for git bisect picking up on non-failing tests due to output truncation
- fix edge case where success status hash was being updated when one of the ci matrices succeeded

### CODE REVIEW

File: .github/automation/aarch64/test.sh [hunk 1]
```diff
@@ -32,5 +32,5 @@ if [[ "$ONEDNN_THREADING" == "SEQ" ]]; then
 fi
 
 set -x
-ctest --no-tests=error --output-on-failure -E $("${SCRIPT_DIR}"/skipped-tests.sh) --output-junit $1
+ctest --no-tests=error --output-on-failure -E $("${SCRIPT_DIR}"/skipped-tests.sh) --output-junit $1 -O $2; cat $2
```
Comment (Sqvid) [inline .github/automation/aarch64/test.sh]: Assign `$1`, and `$2` to a well-named variables so it is obvious what they contain.

Why do you cat `$2` at the end? Doesn't `ctest` write to `stdout` even if you provide `-O`?

Comment (Ryo-not-rio) [reply .github/automation/aarch64/test.sh]: Done, if you look at the output here: https://github.com/uxlfoundation/oneDNN/actions/runs/18282199126/job/52048351701 you can see that there's only one output, using `cat`

Comment (Sqvid) [reply .github/automation/aarch64/test.sh]: Not sure why the behavior seems different on my system but is the content of the file passed to `-O <file>` supposed to be the same as what would normally be printed to `stdout`? In other words are your trying to redirect the output to a file and print it? In which case you probably just want to [pipe it into `tee <file>`](https://man7.org/linux/man-pages/man1/tee.1.html).

Comment (Ryo-not-rio) [reply .github/automation/aarch64/test.sh]: done

**Review by Sqvid — APPROVED** [review]
